javascript/src/types.ts

```diff
@@ -37,6 +37,7 @@ export type Doc<T> = { readonly [P in keyof T]: T[P] }
 export type PatchInfo<T> = {
   before: Doc<T>
   after: Doc<T>
+  source: string
```

Comment:
Is it worth adding a `Source` type here which is a union of the possible sources? Something like
```ts
export type PatchSource = "from" | "change" | "changeAt" | "merge" | "change" | "loadIncremental" | "change" | "sync";
```
Could add it as the type on `progressDocument` too?